docs/troubleshooting-guide/sql-error-message.md

```diff
@@ -0,0 +1,199 @@
+## What's wrong with your SQL query?
+
+- [I get an error message when I run my SQL query][sql-debugging].
+- [I don't have an error message, but my query results are incorrect][troubleshoot-sql-logic].
+
+## Debugging SQL queries
+
+If your SQL query contains [SQL variables][sql-variable-def] that look like `{% raw %}{{ variable }}{% endraw %}`, go to [Troubleshooting SQL variables][troubleshoot-sql-variables] first.
+
+1. Go to the line that is failing in your SQL query.
+   - [I don’t know where my SQL query is failing][how-to-find-failing-sql-line].
+2. Check the [SQL syntax][troubleshoot-sql-syntax] on the line that is failing in your SQL query.
+3. Check your [query logic][troubleshoot-sql-logic] if the query uses joins, subqueries, or CTEs.
+
+### How does SQL debugging work?
```

Comment:
Should we encourage people to limit their results when debugging to speed up query round trips?